From 24df4f1fc800f11e44c44f8036e7d8d7ee90b62a Mon Sep 17 00:00:00 2001
From: Alx Sa <cmyk.student@gmail.com>
Date: Tue, 23 Jul 2024 21:21:53 +0000
Subject: [PATCH] plug-ins: Backport of fixes for gcc-14

This backports a2458f15 and 51311f91
for 2.10 so that it can build with
gcc-14.
---
 plug-ins/file-tiff/file-tiff-load.c | 4 ++--
 plug-ins/metadata/metadata-editor.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/plug-ins/file-tiff/file-tiff-load.c b/plug-ins/file-tiff/file-tiff-load.c
index 72e9538032..4fc3b6c75f 100644
--- a/plug-ins/file-tiff/file-tiff-load.c
+++ b/plug-ins/file-tiff/file-tiff-load.c
@@ -1301,8 +1301,8 @@ load_image (GFile        *file,
 
       /* any resolution info in the file? */
       {
-        gfloat   xres = 72.0;
-        gfloat   yres = 72.0;
+        gdouble  xres = 72.0;
+        gdouble  yres = 72.0;
         gushort  read_unit;
         GimpUnit unit = GIMP_UNIT_PIXEL; /* invalid unit */
 
diff --git a/plug-ins/metadata/metadata-editor.c b/plug-ins/metadata/metadata-editor.c
index 87a2e87098..9e8a08a177 100644
--- a/plug-ins/metadata/metadata-editor.c
+++ b/plug-ins/metadata/metadata-editor.c
@@ -2140,7 +2140,7 @@ metadata_dialog_editor_set_metadata (GExiv2Metadata *metadata,
                             }
                           else
                             {
-                              if (! g_strv_contains (values, equiv_values[evi]))
+                              if (! g_strv_contains ((const gchar * const *) values, equiv_values[evi]))
                                 {
                                   gchar *tmpvalue;
 
-- 
2.46.0

